libraries/AP_Notify/Display.cpp

```diff
@@ -404,15 +404,46 @@ void Display::update()
 
 void Display::update_all()
 {
-    update_text(0);
-    update_mode(1);
-    update_battery(2);
+    if(!BIT_IS_SET(pNotify->_send_text_src_override, 0)) {
+        update_text(0);
+    }
+
+    if(!BIT_IS_SET(pNotify->_send_text_src_override, 1)) {
+        update_mode(1);
+    }
+
+    if(!BIT_IS_SET(pNotify->_send_text_src_override, 2)) {
+        update_battery(2);
+    }
+
 #if AP_GPS_ENABLED
-    update_gps(3);
+    if(!BIT_IS_SET(pNotify->_send_text_src_override, 3)) {
+        update_gps(3);
+    }
 #endif
-    //update_gps_sats(4);
-    update_prearm(4);
-    update_ekf(5);
+
+    if(!BIT_IS_SET(pNotify->_send_text_src_override, 4)) {
+        //update_gps_sats(4);
+        update_prearm(4);
+    }
+
+    if(!BIT_IS_SET(pNotify->_send_text_src_override, 5)) {
+        update_ekf(5);
+    }
+}
+
+void Display::send_text(const char *text, uint8_t r)
+{
+    if (nullptr == text) {
+        return;
+    }
+    if (r >= NOTIFY_TEXT_NUM_ROWS) {
+        return;
+    }
+    char txt [NOTIFY_TEXT_BUFFER_SIZE] = {};
```

Comment:
I would have expected `DISPLAY_MESSAGE_SIZE` does it auto scroll the longer string for us?

And I guess if it does do we want that? Or should the script writer do there own scrolling?

Reply:
Yup, this was wrong, copied it from Display:update_text which did scroll. Fixed in https://github.com/ArduPilot/ardupilot/pull/26066/commits/6fbe3cd76772be8d0b36c1b23cd0a56a93b0db9e